File: 704.binary-search.cs
/*
 * @lc app=leetcode id=704 lang=csharp
 *
 * [704] Binary Search
 */

// @lc code=start
using System.Collections.Generic;
using System;
public class Solution 
{
    public static int bisectLeft(int[] arr, int x)
    {
        int lo = 0;
        int hi = arr.Length;
        while (lo < hi)
        {
            int mid = lo + (hi - lo) / 2;
            if (arr[mid] < x)
            {
                lo = mid + 1;
            }
            else
            {
                hi = mid;
            }
        }
        return lo;     
    }
    public int Search(int[] nums, int target) 
    {
        int index = bisectLeft(nums, target);
        return index == nums.Length || nums[index] != target ? -1 : index;
    }
}
// @lc code=end

File: 704.binary-search.py
#
# @lc app=leetcode id=704 lang=python3
#
# [704] Binary Search
#

# @lc code=start
import bisect
from typing import List


class Solution:
    def search(self, nums: List[int], target: int) -> int:
        index = bisect.bisect_left(nums, target)
        return -1 if index == len(nums) or nums[index] != target else index 
# @lc code=end

